tools/internal_ci/helper_scripts/prepare_build_linux_rc

```diff
@@ -18,6 +18,9 @@
 # Need to increase open files limit for c tests
 ulimit -n 32768
 
+# This is required by the android_ndk_repository repo rule.
+export ANDROID_NDK_HOME=$ANDROID_HOME/ndk-bundle
```

Comment:
Does this mean that unless the ANDROID_NDK_HOME variable is set, bazel 5 build is going to fail?
that seems pretty rought. 
- does the android NDK actually have to be present? or it's just that that env variable needs to be set. If it actually needs to be installed, that's even worse.
- do we really set this env variable for ALL our linux builds?

Also, it looks like this hack won't cover the situations where we run bazel build under docker (e.g. bazel distribtests).  We should probably add a distribtest for bazel 5 ASAP.

Reply:
Thanks for the explanation.
Let's move the export ANDROID_NDK_HOME out of `prepare_build_linux_rc` and only add it to the jobs that actually need to build android.

I think the affected jobs would be 
grpc/core/master/linux/grpc_binder_transport_apk

and perhaps also 
grpc/core/master/linux/grpc_android